src/routes/manage_job/get.rs

```diff
@@ -47,6 +73,338 @@ pub struct JobStatusResponseBody {
+				"is_reachable" => {
+					is_reachable = val
+						.as_str()
+						.ok_or("is_reachable should be a string")?
+						.to_string()
+				}
+				"misc" => {
+					let misc_obj = val.as_object().ok_or("misc field should be an object")?;
+					for (key, val) in misc_obj.keys().zip(misc_obj.values()) {
+						match key.as_str() {
+							"error" => error = Some(val.to_string()),
+							"is_disposable" => {
+								misc_is_disposable =
+									val.as_bool().ok_or("is_disposable should be a boolean")?
+							}
+							"is_role_account" => {
+								misc_is_role_account =
+									val.as_bool().ok_or("is_role_account should be a boolean")?
+							}
+							_ => {}
+						}
+					}
+				}
+				"mx" => {
+					let mx_obj = val.as_object().ok_or("mx field should be an object")?;
+					for (key, val) in mx_obj.keys().zip(mx_obj.values()) {
+						match key.as_str() {
+							"error" => error = Some(val.to_string()),
+							"accepts_email" => {
+								mx_accepts_mail =
+									val.as_bool().ok_or("accepts_email should be a boolean")?
+							}
+							_ => {}
+						}
+					}
+				}
+				"smtp" => {
+					let smtp_obj = val.as_object().ok_or("mx field should be an object")?;
+					for (key, val) in smtp_obj.keys().zip(smtp_obj.values()) {
+						match key.as_str() {
+							"error" => error = Some(val.to_string()),
+							"can_connect_smtp" => {
+								smtp_can_connect = val
+									.as_bool()
+									.ok_or("can_connect_smtp should be a boolean")?
+							}
+							"has_full_inbox" => {
+								smtp_has_full_inbox =
+									val.as_bool().ok_or("has_full_inbox should be a boolean")?
+							}
+							"is_catch_all" => {
+								smtp_is_catch_all =
+									val.as_bool().ok_or("is_catch_all should be a boolean")?
+							}
+							"is_deliverable" => {
+								smtp_is_deliverable =
+									val.as_bool().ok_or("is_deliverable should be a boolean")?
+							}
+							"is_disabled" => {
+								smtp_is_disabled =
+									val.as_bool().ok_or("is_disabled should be a boolean")?
+							}
+							_ => {}
+						}
+					}
+				}
+				"syntax" => {
+					let syntax_obj = val.as_object().ok_or("syntax field should be an object")?;
+					for (key, val) in syntax_obj.keys().zip(syntax_obj.values()) {
+						match key.as_str() {
+							"error" => error = Some(val.to_string()),
+							"is_valid_syntax" => {
+								syntax_is_valid_syntax =
+									val.as_bool().ok_or("is_valid_syntax should be a boolean")?
+							}
+							"username" => {
+								syntax_username = val
+									.as_str()
+									.ok_or("username should be a string")?
+									.to_string()
+							}
+							"domain" => {
+								syntax_domain =
+									val.as_str().ok_or("domain should be a string")?.to_string()
+							}
+							_ => {}
+						}
+					}
+				}
+				// ignore unknown fields
+				_ => {}
+			}
+		}
+
+		Ok(JobResultCsvResponse {
+			input,
+			is_reachable,
+			misc_is_disposable,
+			misc_is_role_account,
+			mx_accepts_mail,
```

Comment:
how hard would it be to transform the CSV columns from `mx_accepts_mail` to `mx.accepts_mail`?

Reply:
Struct fields can't have `.` in their names, I've done it using the `serde(rename = "serialize_name")`.